tracer/src/Datadog.Trace/Iast/Analyzers/HardcodedSecretsAnalyzer.cs

```diff
@@ -94,12 +92,11 @@ private void PoolingThread()
                     }
                 }
 
-                _waitEvent.Wait(2_000);
+                await Task.Delay(2000, cancellationToken).ConfigureAwait(false);

```

Comment:
Personally I find the use of underscore for a thousand separator a nice touch, but hardly worth a new CI run.

Reply:
I had to refresh the branch anyway, so I added the change.